python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -115,17 +116,44 @@ class Meta:
         name = "BulkActionStatus"
 
 
+class GrapheneAssetBackfillTargetPartitions(graphene.ObjectType):
+    class Meta:
+        name = "AssetBackfillTargetPartitions"
+
+    ranges = graphene.List(
+        graphene.NonNull("dagster_graphql.schema.partition_sets.GraphenePartitionKeyRange")
+    )
+    partitionKeys = graphene.List(graphene.NonNull(graphene.String))
+
+    def __init__(self, partition_subset: PartitionsSubset):
+        from dagster_graphql.schema.partition_sets import GraphenePartitionKeyRange
+
+        if isinstance(partition_subset, TimeWindowPartitionsSubset):
+            ranges = [
+                GraphenePartitionKeyRange(start, end)
+                for start, end in partition_subset.get_partition_key_ranges()
+            ]
+            partition_keys = None
+        else:  # Default partitions subset
+            ranges = None
+            partition_keys = partition_subset.get_partition_keys()
+
+        super().__init__(
+            ranges=ranges,
+            partitionKeys=partition_keys,
+        )
+
+
 class GrapheneAssetBackfillData(graphene.ObjectType):
     class Meta:
         name = "AssetBackfillData"
 
     assetBackfillStatuses = non_null_list(
         "dagster_graphql.schema.partition_sets.GrapheneAssetBackfillStatus"
     )
-    rootAssetTargetedRanges = graphene.List(
-        graphene.NonNull("dagster_graphql.schema.partition_sets.GraphenePartitionKeyRange")
+    rootTargetedPartitions = graphene.NonNull(
```

Comment:
it is possible for this to be null right? when the asset backfill data can't be deserialized

Reply:
ah nevermind -- thought this was `PartitionBackfill`